flik/linesearch/conditions.py

```diff
@@ -1 +1,47 @@
 """Conditions for the line search algorithm."""
+def soft_wolfe(gradf, valf, alpha, direction, const2=0.9):
+    """Curvature condition
+
+    Parameters
+    ----------
+    gradf: function
+        Gradient of objective nonlinear function
```

Comment:
It would be better to use `callable` instead of `function`, I think. Also, it would be good to include the input and the output of the function somewhere.